fdbclient/DatabaseContext.h

```diff
@@ -434,13 +437,23 @@ class DatabaseContext : public ReferenceCounted<DatabaseContext>, public FastAll
 	// Cache of the latest commit versions of storage servers.
 	VersionVector ssVersionVectorCache;
 
-        // Adds or updates the specified (SS, TSS) pair in the TSS mapping (if not already present).
-        // Requests to the storage server will be duplicated to the TSS.
+	// Adds or updates the specified (SS, TSS) pair in the TSS mapping (if not already present).
+	// Requests to the storage server will be duplicated to the TSS.
 	void addTssMapping(StorageServerInterface const& ssi, StorageServerInterface const& tssi);
 
-        // Removes the storage server and its TSS pair from the TSS mapping (if present).
-        // Requests to the storage server will no longer be duplicated to its pair TSS.
+	// Removes the storage server and its TSS pair from the TSS mapping (if present).
+	// Requests to the storage server will no longer be duplicated to its pair TSS.
 	void removeTssMapping(StorageServerInterface const& ssi);
+
+	// Adds or updates the specified (UID, Tag) pair in the tag mapping.
+	void addSSIdTagMapping(const UID& uid, const Tag& tag);
+
+	// Returns the latest commit versions that mutated the specified storage servers
+	/// @note returns the latest commit version for a storage server only if the latest
+	// commit version of that storage server is below the specified "readVersion".
```

Comment:
Does a client ever do reads at a version that is below the latest commit version in the database? If so, we won't need to do the filtering based on "readVersion".

Reply:
I thought a client is allowed to `set_read_version` ([link](https://apple.github.io/foundationdb/api-python.html#versions)) , so could choose a version below the last commit version.

Reply:
Ok, thanks for that info!